core/controllers/cron_test.py

```diff
@@ -447,6 +447,241 @@ def test_email_not_sent_if_reviewer_ids_is_empty(self) -> None:
         self.logout()
 
 
+class CronMailReviewerNewSuggestionsHandlerTests(
+        test_utils.GenericTestBase):
+
+    target_id = 'exp1'
+    language_code = 'en'
+    default_translation_html = '<p>Sample translation</p>'
+    AUTHOR_USERNAME: Final = 'author'
+    AUTHOR_EMAIL: Final = 'author@example.com'
+    REVIEWER_USERNAME: Final = 'reviewer'
+    REVIEWER_EMAIL: Final = 'reviewer@community.org'
+
+    def _mock_send_contributor_dashboard_reviewers_emails(
```

Comment:
Can you avoid setting test values in self? You are adding state to your tests when they should be stateless.

Reply:
I took reference from existing tests, for e.g https://github.com/oppia/oppia/pull/19045/files#diff-da20f075b7775a03b83442f138f0b1a7038ba690db8c102f543528576f534598R326.

Reply:
That should be changed too. If it's an immutable value that is used across the test then it's okay to assign it to self. Otherwise, each test should not be setting distince values to self.

Reply:
Okay but this method is specifically designed to be used as a replacement for email_manager.send_reviewer_notifications during testing.
 
So I think in this case, since the method is indeed intended to be used across multiple tests, storing the relevant state (like self.reviewer_ids_by_language and self.reviewable_suggestions_by_language) as attributes of the test class (self) is appropriate. WDYT ?

Reply:
I would say spy on the mocked method and then assert that the method was called with particular arguments. The problem with the current approach is that if you run tests in parallel and all are setting up values on the same object (i.e. `self`), the value on self might change before you've had a chance to assert the result.

Reply:
If you replace the method with mock objects from python's unittest libraries, you can use assert_called_with, assert_called_once_with or call_args_list to assert that the mock was called with the right arguments

Reply:
Done, thanks.